### PR TITLE
Simplify Tina wrapper

### LINKED ISSUE #2545: Move default Tina wrapper to its own file
## Summary

One clunky part of setting up Tina is handling the dynamic imports. 

Here's the most simple example of setting up Tina in your _app.jsx
```jsx
// pages/_app.js
import dynamic from 'next/dynamic'
import { TinaEditProvider } from 'tinacms/dist/edit-state'
const TinaCMS = dynamic(() => import('tinacms'), { ssr: false })

const App = ({ Component, pageProps }) => {
  return (
    <>
      <TinaEditProvider
        editMode={
          <TinaCMS apiURL="http://localhost:4001/graphql">
            <Component {...pageProps} />
          </TinaCMS>
        }
      >
        <Component {...pageProps} />
      </TinaEditProvider>
    </>
  )
}

export default App
```

Our `tina init` will ask if you want to overwrite your _app.jsx file (which I think would think for 99% of NextJS sites, that wouldn't be ideal). If we could instead output all the of the Tina setup in its own file, then Tina could be added by running tina init, and wrapping the layout in one simple `<Tina>` wrapper component.

We kind of touched on this in [this thread](https://github.com/tinacms/tinacms/discussions/2402), but I don't see a reason that it couldn't be tackled independently of the "Extending Tina" work
 
### Basic example

The cli would create `./components/tina.(jsx|tsx)`  
```jsx
import dynamic from "next/dynamic";
const TinaCMS = dynamic(() => import("tinacms"), { ssr: false });
import { TinaEditProvider } from "tinacms/dist/edit-state";

const branch = "main";
// When working locally, hit our local filesystem.
// On a Vercel deployment, hit the Tina Cloud API
const apiURL =
  process.env.NODE_ENV == "development"
    ? "http://localhost:4001/graphql"
    : `https://content.tinajs.io/content/${process.env.NEXT_PUBLIC_TINA_CLIENT_ID}/github/${branch}`;

const Tina = ({ children }) => {
  return (
    <>
      <TinaEditProvider
        showEditButton={false}
        editMode={
          <TinaCMS
            apiURL={apiURL}
            cmsCallback={(cms) => {
              cms.flags.set("tina-admin", true);
            }}
          >
            {children}
          </TinaCMS>
        }
      >
        {children}
      </TinaEditProvider>
    </>
  );
};

export default Tina;
```
^ This would only ever need to be adjusted later by the user if they want to update their config.

The CLI would instruct the user to add `<TinaCMS>` to their _app.jsx

```jsx
import TinaCMS from "../components/Tina";

const App = ({ Component, pageProps }) => {
  return (
    <>
      <TinaCMS>
        <Component {...pageProps} />
      </TinaCMS>
    </>
  );
};

export default App;
```

### Motivation

Now the user doesn't need to manually setup the dynamic imports. 
It's also less error-prone, as before in the user's _app.jsx, there was 2 places where the child `Component` was rendered (one in edit-mode, 1 in production):
```jsx
   <>
      <TinaEditProvider
        showEditButton={true}
        editMode={
          <TinaCMS apiURL={apiURL}>
            <Component {...pageProps} />
          </TinaCMS>
        }
      >
        <Component {...pageProps} />
      </TinaEditProvider>
    </>
```
If the user didn't pass those `pageProps` to one of the children components, it would cause issues. Now, it's only handled in one place in the user's _app.jsx:

```jsx
      <TinaCMS>
        <Component {...pageProps} />
      </TinaCMS>
    </>
```

This project would entail:
- Updating the CLI
- Updating the docs
- Updating our examples

---
Note: See[ this draft PR](https://github.com/tinacms/tinacms.org/pull/1226) for a list of remaining TODOs

### PR DESCRIPTION
**Pull request by jamespohalloran**
fixes #2545

## New TinaProvider Pattern

Previously, wrapping Tina around a site looked like this:

```jsx
// pages/_app.js
import dynamic from 'next/dynamic'
import { TinaEditProvider } from 'tinacms/dist/edit-state'
const TinaCMS = dynamic(() => import('tinacms'), { ssr: false })

const App = ({ Component, pageProps }) => {
  return (
    <>
      <TinaEditProvider
        editMode={
          <TinaCMS apiURL="http://localhost:4001/graphql">
            <Component {...pageProps} />
          </TinaCMS>
        }
      >
        <Component {...pageProps} />
      </TinaEditProvider>
    </>
  )
}

export default App
```

In order to make sure Tina was code-split from the production bundle, The user needed to explictly handle dynamically importing Tina, and implementing both the TinaEditProvider, and the Tina provider.

To make this more foolproof for the devs, we've introduced a new pattern for wrapping the site with Tina.

the `@tinacms/cli init` script now creates the following new files:

- .tina/components/TinaProvider
- .tina/components/DynamicTinaProvider

`DynamicTinaProvider` now handles all of the code-splitting under the hood for you, so your app.jsx will now look like:

```jsx
import TinaCMS from '../.tina/DynamicTinaProvider'

const App = ({ Component, pageProps }) => {
  return (
    <>
      <TinaCMS>
        <Component {...pageProps} />
      </TinaCMS>
    </>
  )
}

export default App
```

With this pattern, by default Tina can now be extended from `.tina/schema.ts`

```tsx
import { defineConfig } from 'tinacms'

// ...

export defineConfig(
{
	apiURL: 'http://localhost:'
})
```

This change is completely backwards compatible and is currently an optional pattern for existing users. You may choose to keep everything contained in the _app.jsx file!

**Note: If you are upgrading to this pattern from an existing Tina setup, you will need to change your `defineSchema`, now that this file is imported from the frontend.

```diff
- import { defineSchema } from '@tinacms/cli'
+ import { defineSchema } from 'tinacms'
```

Some docs subtasks:

- [x] Add new setup instructions to ["Getting Started"](https://github.com/tinacms/tinacms.org/pull/1226/files#diff-e9ba65fc7ecbd5fc0567191f02c97642e8b92f56960b90486067977c966c0c83) page.
- [x] Update [cloudinary docs](https://tina.io/docs/media-cloudinary/)
- [x] Update [markdown docs](https://tina.io/docs/editing/markdown/)
- [x] Update [CLI reference](https://tina.io/docs/cli-overview/) to changes the _app.jsx references to the files in the new .tina directory
- [x] Update https://tina.io/docs/advanced/customizing-forms/ (this is no longer configured in _app.jsx)
- [x] Update [extending tina](https://tina.io/docs/advanced/extending-tina/)
- [x] Update [connecting the site](https://tina.io/docs/tina-cloud/connecting-site/)
- [x] Remove [dated guide](https://tina.io/guides/tina-cloud/add-tinacms-to-existing-site/overview/)
- [ ] Update the starters

### CODE REVIEW

**Comment by spbyrne** [comment]
I don't know the technical pros/cons but this looks slick 👍

**Comment by jeffsee55** [comment]
My only concern is mixing up our messaging with this and then we go fully towards the extending Tina patterns and things have to change again. Either way, I'm a fan just not sure on timing.

**Comment by jamespohalloran** [comment]
> My only concern is mixing up our messaging with this and then we go fully towards the extending Tina patterns and things have to change again. 

What do you foresee changing with the extending Tina work? I think there will be lots of changes to the user's ]Tina Provider](https://github.com/tinacms/tinacms/pull/2549/files#diff-25ed644f964c116fb449f618ec00bc35b9428179ae2e3068fbcf0846644229feR13) component, regardless if it lives in a _app.jsx, or a .tina/... file

My main reasoning for wanting to tackle it sooner rather than later was that so much of the docs onboarding are changing with the admin work, that it might make sense to include this new way of setting things up at the same time

**Comment by jamespohalloran** [comment]
> Mainly just the location of this stuff, will the `defineSchema` go in the same file as the `TinaCMS` component? And will all of that live in `.tina/some-file`?

IMO, I think the "Extending Tina" pitch is a bit too broad as is, and shouldn't block this. This is a pretty minimal change for us, and the risk of moving/renaming a file or 2 shouldn't be too impactful for us or the user. (This is also an opt-in pattern for current users)

**Comment by jamespohalloran** [comment]
Just outlined (in the issue body) some docs-related sub-tasks needed to move this PR forward.

**Comment by jamespohalloran** [comment]
So right now, the CLI outputs the following new files:
- [/.tina/components/provider/_TinaProvider.jsx](https://github.com/tinacms/tinacms/pull/2549/files#diff-25ed644f964c116fb449f618ec00bc35b9428179ae2e3068fbcf0846644229fe)
- [/.tina/components/provider/index.jsx](https://github.com/tinacms/tinacms/pull/2549/files#diff-1c299f7d9a3d3b35d120fc2d4520d1557318886b66b3c23f580ccd52a64bd3cd)

I think we need to confirm the following before we can move forward:
- [ ] Are we okay with those file paths? Or does any Extending Tina work potentially block this?
- [ ] Are we okay saying that for now, the user needs to manually add the `TinaCMS` container (we're not going to add it to the admin)

Are there any other concerns blocking this? If not we should be able to finish off the remaining documentation tasks in this PR description and mark this as ready for review @jeffsee55 @logan-anderson

**Comment by jeffsee55** [comment]
IMO the extra `provider` folder doesn't convey much meaning, just putting them in `components` seems fine.

> Are we okay saying that for now, the user needs to manually add the TinaCMS container (we're not going to add it to the admin)

Can we still add it _if_ there's no `_app.js`? Otherwise, yeah seems like a simple enough thing to require.

**Comment by jamespohalloran** [comment]
> IMO the extra `provider` folder doesn't convey much meaning, just putting them in `components` seems fine.
> 

Any suggestions? My main reasoning was that the main `_TinaProvider.jsx` should never be imported directly, outside of that directory. 
_Not that they couldn't still reach for it with `./.tina/provider/_TinaProvider.jsx`__, but it adds an extra layer of encapsulation_

**Comment by logan-anderson** [comment]
A couple of suggestions for the tina configuration file (The one users should not import)
- TinaEditConfig
- TinaConfig
- _{One of the above}

For the one that the user does import we are thinking `DynamicTinaProvider` and we will have them in a flat file structor in `.tina/components`

**Comment by jeffsee55** [comment]
Don't let my comment stop you @logan-anderson, whatever you think works for me

**Comment by jamespohalloran** [comment]
> A couple of suggestions for the tina configuration file (The one users should not import)
> 
> * TinaEditConfig
> * TinaConfig
> * _{One of the above}
> 
> For the one that the user does import we are thinking `DynamicTinaProvider` and we will have them in a flat file structor in `.tina/components`

I'm going to vote for 
- TinaConfig (File the user edits)
- DynamicTinaProvider (File the user imports, to wrap the layout)

I'm good with any of the options though

File: packages/@tinacms/cli/src/cmds/init/index.ts [hunk 1]
```diff
@@ -95,13 +95,13 @@ export async function tinaSetup(ctx: any, next: () => void, options) {
...
+    !fs.pathExistsSync(TinaFolder) &&
     !fs.existsSync(TinaProviderPath) &&
-    !fs.existsSync(TinaProviderPathIndex)
+    !fs.existsSync(TinaDynamicProvider)
   ) {
-    fs.mkdirpSync(TinaProviderFolder)
+    fs.mkdirpSync(TinaFolder)
     fs.writeFileSync(TinaProviderPath, TinaProvider)
-    fs.writeFileSync(TinaProviderPathIndex, TinaProviderIndex)
+    fs.writeFileSync(TinaDynamicProvider, TinaProviderDynamic)
```
Comment (jamespohalloran) [inline packages/@tinacms/cli/src/cmds/init/index.ts:104]: @logan-anderson You might still be working on this, but I see we sometimes call it `TinaProviderDynamic` & sometimes `TinaDynamicProvider`

**Comment by logan-anderson** [comment]
This is ready for review and this is how it can be tested

https://www.loom.com/share/5d78a4e8a7944555832f7233cca83f60

**Comment by jamespohalloran** [comment]
I opened this PR, so I can't improve (even though @logan-anderson did all the work). But I approve! Looks great to me

**Pull request merged**